### PR TITLE
Check maintenance window time when using WebCron

### PR DESCRIPTION
**Pull request by k-popov**
The patch moves maintenance window check earlier so that
maintenance window was respected when using both system cron
and WebCron. Before the change webcron used to select
**only** time-sensitive tasks.

Signed-off-by: Kirill Popov <kirill.s.popov@gmail.com>

### CODE REVIEW

**Review by nickvergessen — COMMENTED** [review]
Tkme sensitivity should only be considered for cron, not for web or ajax

File: cron.php [hunk 1]
```diff
@@ -167,8 +168,9 @@
 		} else {
 			// Work and success :-)
 			$jobList = \OC::$server->getJobList();
-			$job = $jobList->getNext();
+			$job = $jobList->getNext($onlyTimeSensitive);
```
Comment (nickvergessen) [inline cron.php:171]: The actual fix would be to just inject false here or change the default of the method

**Review by nickvergessen — CHANGES_REQUESTED** [review]

File: cron.php (hunk 1 above)
Comment (k-popov) [reply cron.php:171]: Could you please comment why?
I'm running NextCloud in Kubernetes and thus I can't use system cron daemon. Having Kubernetes' `CronJob` is a solution - it runs periodically and "curl's" `cron.php`. In this setup I still have short background jobs and time-sensitive ones which I want to run during maintenance hours.

**Comment by k-popov** [comment]
I've opened #32109 with argument default change.

File: cron.php (hunk 1 above)
Comment (nickvergessen) [reply cron.php:171]: Explanation in https://github.com/nextcloud/server/issues/30899#issuecomment-1108360345

**Pull request closed without merging**